axelor-account/src/main/java/com/axelor/apps/account/web/AnalyticDistributionLineController.java

```diff
@@ -63,20 +64,31 @@ public void validateLines(ActionRequest request, ActionResponse response) {
   public void manageNewAnalyticDistributionLine(ActionRequest request, ActionResponse response)
       throws AxelorException {
     try {
-      MoveLine moveLine = request.getContext().getParent().asType(MoveLine.class);
-      if (moveLine != null)
+      Context parent = request.getContext().getParent();
+      if (MoveLine.class.toString().equals(parent.getContextClass().toString())) {
+        MoveLine line = parent.asType(MoveLine.class);
         response.setValue(
             "analyticJournal",
             Beans.get(AccountConfigService.class)
-                .getAccountConfig(moveLine.getAccount().getCompany())
+                .getAccountConfig(line.getAccount().getCompany())
                 .getAnalyticJournal());
-      if (moveLine.getDate() != null) {
-        response.setValue("date", moveLine.getDate());
-      } else {
+        if (line.getDate() != null) {
+          response.setValue("date", line.getDate());
+        } else {
+          response.setValue(
+              "date", Beans.get(AppBaseService.class).getTodayDate(line.getAccount().getCompany()));
+        }
+      } else if (InvoiceLine.class.toString().equals(parent.getContextClass().toString())) {
+        InvoiceLine line = request.getContext().getParent().asType(InvoiceLine.class);
         response.setValue(
-            "date",
-            Beans.get(AppBaseService.class).getTodayDate(moveLine.getAccount().getCompany()));
+            "analyticJournal",
+            Beans.get(AccountConfigService.class)
+                .getAccountConfig(line.getAccount().getCompany())
+                .getAnalyticJournal());
+        response.setValue(
+            "date", Beans.get(AppBaseService.class).getTodayDate(line.getAccount().getCompany()));
```

Comment:
Could you create 2 methods in a service, one to get the AnalyticJournal and one to get the date?